### PR TITLE
Auto redirect to last-visited domain

### PR DESCRIPTION
**Pull request by sravfeyn**
http://manage.dimagi.com/default.asp?164618
- Save last-visited-domain to cookie
- When user visits https://www.commcarehq.org/domain/select/ or https://www.commcarehq.org/, redirect to last-visited-domain dashboard (if available, if not no-redirect).
- Do not auto-redirect if there are pending project invitations

@esoergel

### CODE REVIEW

File: corehq/util/global_request/middleware.py [hunk 1]
```diff
@@ -7,4 +7,11 @@ def process_request(self, request):
         set_request(request)
 
     def process_response(self, request, response):
```
Comment (sravfeyn) [inline corehq/util/global_request/middleware.py]: May be one middleware should do only one function, should I make a new separate RememberDomainMiddleware instead?

**Comment by snopoke** [comment]
I think a separate middleware would be better. I also think that using the Django session vs cookies is a better approach: https://docs.djangoproject.com/en/1.8/topics/http/sessions/

**Comment by sravfeyn** [comment]
@snopoke or @esoergel

**Comment by benrudolph** [comment]
this'll be super nice to have. looks good to me and i also like the middleware approach

**Pull request merged**